jwst/cube_build/ifu_cube.py

```diff
@@ -1306,13 +1309,47 @@ def setup_ifucube_wcs(self):
             log.debug(f'Working on data from {this_a}, {this_b}')
             n = len(self.master_table.FileMap[self.instrument][this_a][this_b])
             log.debug('number of files %d', n)
+            
+            # find the median center declination if we have an offset file
+            if self.offsets is not None:
+                decs = []
+                for k in range(n):
+                    input_file = self.master_table.FileMap[self.instrument][this_a][this_b][k]
+                    input_model = datamodels.open(input_file)
+                    spatial_box = input_model.meta.wcsinfo.s_region
+                    s = spatial_box.split(' ')
+                    cb1 = float(s[4])
+                    cb2 = float(s[6])
+                    cb3 = float(s[8])
+                    cb4 = float(s[10])
+                    m = (cb1 + cb2 + cb3 + cb4)/4
+                    decs.append(m)
+
+                self.median_dec = np.nanmedian(decs)
+                # fold in the median_dec information into ra offset
+                noffsets = len(self.offsets['raoffset'])
+                # convert ra and dec offsets to degrees and adjust ra offset for cos(dec) 
+                for im in range(noffsets):
+                    self.offsets['raoffset'][im] = (self.offsets['raoffset'][im]/3600.0)/np.cos(self.median_dec*deg2rad)
+                    self.offsets['decoffset'][im] = self.offsets['decoffset'][im]/3600.0
+                
             for k in range(n):
                 lmin = 0.0
                 lmax = 0.0
 
                 input_file = self.master_table.FileMap[self.instrument][this_a][this_b][k]
                 input_model = datamodels.open(input_file)
-
+# ________________________________________________________________________________
+                # If offsets are provided. Pull in ra and dec offsets.
+                raoffset = 0.0
+                decoffset = 0.0 
+                # pull out ra dec offset if it exists
+                if self.offsets is not None:
+                    filename = input_model.meta.filename
+                    index = self.offsets['filename'].index(filename)
+                    raoffset = self.offsets['raoffset'][index]
+                    decoffset = self.offsets['decoffset'][index]
```

Comment:
Does the schema require that this is always valid?  That is, is it possible to get to this point with a mismatched file and offset list?

Reply:
Not sure I understand the question. If the filename is not in the offset file then it gives an error to the user.
If you are asking if the user has not kept the order of the filenames and the raoffset and decoffsets the same then
that is a problem. But that is why I gave the example of how to make the offset file and it seem fairly straight forward that they need to match. 
Is that  what you are asking ?

Reply:
Sorry, I wasn't clear here either.  My question was - what happens if the user accidentally provides 4 filenames but 3 RA or Dec offset values?  If it gets to this point in the code in that condition, these lines will throw an IndexError.  

I'm not familiar enough with the ASDF validation to be sure, but it didn't look to me like you were checking the length of the lists anywhere and making sure that they match.  Maybe that check just needs to be added to the check_offset_file function?

Reply:
Ok I will see if I can add a check for that

Reply:
I just added it into the code in cube_build as a check.